packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -261,6 +283,46 @@ class GraphQLRequestFactory {
     throw ApiException(
         'Unable to translate the QueryPredicate $queryPredicate to a GraphQL filter.');
   }
+
+  /// Calls `toJson` on the model and removes key/value pairs that refer to
+  /// children when that field is null. The structure of provisioned AppSync `input` type,
+  /// such as `CreateBlogInput` does not include nested types, so they will get
+  /// an error from AppSync if they are included in mutations.
+  ///
+  /// When the model has a parent via a belongsTo, the id from the parent is added
+  /// as a field similar to "blogID" where the value is `post.blog.id`.
+  Map<String, dynamic> buildInputVariableForMutations(Model model) {
+    ModelSchema schema = _getAndValidateSchema(model.getInstanceType(), null);
+    final modelJson = model.toJson();
+
+    // If the model has a parent in the schema, get the ID of parent and field name.
+    String? belongsToModelName; // e.g. "blog"
+    String? belongsToKey; // e.g. "blogID"
+    String? belongsToValue; // the ID value to use from `post.blog.id`
+    final belongsToAssociation = _getBelongsToFieldFromModelSchema(schema);
+    if (belongsToAssociation != null) {
+      belongsToModelName = belongsToAssociation.key;
+      belongsToKey = belongsToAssociation.value.association?.targetName;
+      belongsToValue = modelJson[belongsToModelName]?[idFieldName];
+    }
+
+    // Remove any relational fields with null values.
+    // Non-null values left in place to avoid secret deletion and let AppSync throw.
+    Set<String> associationFields = schema.fields!.entries
+        .where((entry) => entry.value.association != null)
+        .map((entry) => entry.key)
+        .toSet();
+    modelJson.removeWhere((key, dynamic value) =>
```

Comment:
Yeah the difference depends on what we set our native code to expect.  On the native side pretty sure we're checking for the existence of the field in the serialized map and if it's present making a force cast.  If we were to allow null fields, we'd need to check for field existence and then check if the field is null.

Reply:
I looked at this a little more and I think AppSync is correctly interpreting the spec related to input objects regarding this behavior. https://spec.graphql.org/June2018/#sec-Input-Objects "In either case, the input object literal or unordered map must not contain any entries with names not defined by a field of this input object type, otherwise an error must be thrown."